quinn-proto/src/connection/stats.rs

```diff
@@ -0,0 +1,83 @@
+/// Statistics about packets transmitted on a connection
+#[derive(Default)]
+#[non_exhaustive]
+pub struct PacketTransmissionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl PacketTransmissionStats {
+    /// Calculates the average packet size
+    pub fn avg_packet_size(&self) -> u64 {
+        if self.packets != 0 {
+            self.bytes / self.packets
+        } else {
+            0
+        }
+    }
+}
+
+impl std::fmt::Debug for PacketTransmissionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PacketTransmissionStats")
+            .field("packets", &self.packets)
+            .field("bytes", &self.bytes)
+            .field("avg_packet_size", &self.avg_packet_size())
+            .finish()
+    }
+}
+
+/// Statistics about frames transmitted or received on a connection
+#[derive(Default)]
+#[non_exhaustive]
+pub struct FrameCounter {
```

Comment:
I now renamed this, since it seems useful for the sending as well as receiving end